src/config.ts

```diff
@@ -148,7 +151,10 @@ class ConfigurationFactory {
         const url = githubEvent["repository"]
           ? githubEvent["repository"]["url"]
           : `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}`
-        return `${url}/tarball/${ref}`
+        const urlTarball = `/tarball/${ref}`
+        const encodedTarball = encodeURIComponent(urlTarball)
+        const finalTarballUrl = String(url) + String(encodedTarball)
+        return finalTarballUrl
```

Comment:
Same here, consider removing `finalTarballUrl` to avoid unnecessary variable creation.